src/main/java/org/torproject/onionoo/updater/NodeDetailsStatusUpdater.java

```diff
@@ -757,18 +757,39 @@ private void computeEffectiveAndExtendedFamilies() {
   private long startedRdnsLookups = -1L;
 
   private SortedMap<String, String> rdnsLookupResults = new TreeMap<>();
+  private SortedMap<String, List<String>> rdnsVerifiedLookupResults =
+      new TreeMap<>();
+  private SortedMap<String, List<String>> rdnsUnverifiedLookupResults =
+      new TreeMap<>();
 
   private void finishReverseDomainNameLookups() {
     this.reverseDomainNameResolver.finishReverseDomainNameLookups();
     this.startedRdnsLookups =
         this.reverseDomainNameResolver.getLookupStartMillis();
     Map<String, String> lookupResults =
         this.reverseDomainNameResolver.getLookupResults();
+    Map<String, List<String>> verifiedLookupResults =
+        this.reverseDomainNameResolver.getVerifiedLookupResults();
+    Map<String, List<String>> unverifiedLookupResults =
+        this.reverseDomainNameResolver.getUnverifiedLookupResults();
     for (String fingerprint : this.currentRelays) {
       NodeStatus nodeStatus = this.knownNodes.get(fingerprint);
       String hostName = lookupResults.get(nodeStatus.getAddress());
+      List<String> verifiedHostNames =
+              verifiedLookupResults.get(nodeStatus.getAddress());
+      List<String> unverifiedHostNames =
+              unverifiedLookupResults.get(nodeStatus.getAddress());
       if (hostName != null) {
         this.rdnsLookupResults.put(fingerprint, hostName);
+      } else {
+        /* Maintains bug compatibility with previous implementation */
+        this.rdnsLookupResults.put(fingerprint, nodeStatus.getAddress());
+      }
+      if (verifiedHostNames != null && verifiedHostNames.size() > 0) {
```

Comment:
See above, please rewrite as `null != verifiedHostNames`, mostly for consistency.

Reply:
Ah, and rather than checking `xy.size() > 0`, simply check `!xy.isEmpty()`. Easier to read.